src/common/src/array/data_chunk.rs

```diff
@@ -95,23 +96,17 @@ impl DataChunk {
     }
 
     /// Build a `DataChunk` with rows.
+    ///
+    /// Panics if the `rows` is empty.
     pub fn from_rows(rows: &[impl Row], data_types: &[DataType]) -> Self {
-        let mut array_builders = data_types
-            .iter()
-            .map(|data_type| data_type.create_array_builder(1))
-            .collect::<Vec<_>>();
+        let mut builder = DataChunkBuilder::new(data_types.to_vec(), rows.len() + 1);
```

Comment:
Why `+1` here?

Reply:
For code simplicity, we can consume the builder after the loop. Otherwise, it'll output on the last cycle. Not a big deal.

Reply:
```suggestion
        // `append_one_row` will cause the builder to finish immediately once capacity is met.
        // Hence, we allocate an extra row (rows.len() + 1), to avoid the builder finishing prematurely.
        // This just makes the code cleaner, since we can loop through all rows, and consume it finally.
        let mut builder = DataChunkBuilder::new(data_types.to_vec(), rows.len() + 1);
```

Reply:
We may have something like `DataChunkBuilder::new_unlimited` to allow manually consuming chunks

Reply:
Exactly! Actually I'm considering adding a new interface of `new_unlimited(initial_capacity: usize)` to decouple the output chunk size and memory reservation for array builders, so that the chunk size can be `usize::MAX`. Maybe in next PRs when we're going to unify the interfaces for `Stream` and `Data`.